dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/HdfsExtractionTask.java

```diff
@@ -74,21 +83,38 @@ protected Void doRun(TaskRunContext context, @Nonnull ByteSink sink, @Nonnull Ha
     DistributedFileSystem fs = ((HdfsHandle) handle).getDfs();
     // Create a dedicated ExecutorService to use:
     ExecutorService execService =
-        ExecutorManager.newExecutorServiceWithBackpressure("hdfs-extraction", poolSize);
+        ExecutorManager.newExecutorServiceWithBackpressure("hdfs-extraction", threadPoolSize);
     try (Writer output = sink.asCharSink(UTF_8).openBufferedStream();
-        ScanContext scanCtx = new ScanContext(fs, output);
-        ExecutorManager execManager = new ExecutorManager(execService)) {
+        ExecutorManager execManager = new ExecutorManager(execService);
+        ScanContext scanCtx = new ScanContext(execManager, fs, output)) {
 
-      String hdfsPath = "/";
-      FileStatus rootDir = fs.getFileStatus(new Path(hdfsPath));
-      SingleDirScanJob rootJob = new SingleDirScanJob(scanCtx, execManager, rootDir);
-      execManager.execute(rootJob); // The root job executes immediately
+      LOG.info(
+          "Running HDFS extraction\n\t{}: {}\n\t{}: {}",
+          OPT_HDFS_SCAN_ROOT_PATH,
+          hdfsScanRootPath,
+          OPT_THREAD_POOL_SIZE,
+          threadPoolSize);
+      FileStatus rootDir = fs.getFileStatus(new Path(hdfsScanRootPath));
+      scanCtx.submitRootDirScanJob(rootDir, getContentSummaryFor(fs, rootDir));
       execManager.await(); // Wait until all (recursive) tasks are done executing
-      LOG.info(scanCtx.getFormattedStats());
+      LOG.info("Final stats:\n{}", scanCtx.getDetailedStats());
     } finally {
       // Shutdown the dedicated ExecutorService:
       MoreExecutors.shutdownAndAwaitTermination(execService, 100, TimeUnit.MILLISECONDS);
     }
     return null;
   }
+
+  private ContentSummary getContentSummaryFor(DistributedFileSystem dfs, FileStatus file) {
+    try {
+      return dfs.getContentSummary(file.getPath());
+    } catch (org.apache.hadoop.security.AccessControlException exn) {
+      LOG.error(
```

Comment:
I think it should be warning level. The main flow is not broken, but log messages will be less informative.